replication/binlogsyncer.go

```diff
@@ -616,10 +616,8 @@ func (b *BinlogSyncer) writeRegisterSlaveCommand() error {
 	n = copy(data[pos:], b.cfg.User)
 	pos += n
 
-	data[pos] = uint8(len(b.cfg.Password))
+	data[pos] = uint8(0)
 	pos++
-	n = copy(data[pos:], b.cfg.Password)
```

Comment:
😂 Don't know why we bother to send password before. Does different version of server has different behaviour?

Will take a look soon

Reply:
Oh interesting, we don't need to send the password?

Reply:
The password is *only* for use in the output of `SHOW REPLICAS` if `--show-replica-auth-info` is enabled. This makes it a bit of a security hazard and with the length limitation also quite useless.